frontend/apps/reader/modules/readerbookmark.lua

```diff
@@ -61,6 +63,68 @@ function ReaderBookmark:addToMainMenu(menu_items)
     end
 end
 
+function ReaderBookmark:search(query, bookmarks, current_bookmark, bm_menu, viewer_instance, direction)
+    self.search_value = query
+    local content
+    local found_index = 0
+    local start
+    local use_second_loop = true
+    if direction == "forward" then
+        start = current_bookmark + 1
+        if start > #bookmarks then
+            start = 1
+            use_second_loop = false
+        end
+        for i = start, #bookmarks do
+            content = string.lower(bookmarks[i].notes)
+            if content:match(query) then
+                found_index = i
+                break
+            end
+        end
+        if use_second_loop and found_index == 0 then
+            for i = 1, current_bookmark do
+                content = string.lower(bookmarks[i].notes)
+                if content:match(query) then
+                    found_index = i
+                    break
+                end
+            end
+        end
+    -- direction == "previous":
+    else
+        start = current_bookmark - 1
+        if start < 1 then
+            start = #bookmarks
+            use_second_loop = false
+        end
+        for i = start, 1, -1 do
+            content = string.lower(bookmarks[i].notes)
+            if content:match(query) then
+                found_index = i
+                break
+            end
+        end
+        if use_second_loop and found_index == 0 then
+            for i = #bookmarks, current_bookmark, -1 do
+                content = string.lower(bookmarks[i].notes)
+                if content:match(query) then
+                    found_index = i
+                    break
+                end
+            end
+        end
+    end
+
+    if found_index > 0 then
+        UIManager:close(viewer_instance)
+        bm_menu:onMenuHold(bookmarks[found_index])
+        UIManager:show(InfoMessage:new { text = _("Found"), timeout = 1 })
+    else
+        UIManager:show(InfoMessage:new { text = T(_('No bookmarks found with query "%1"'), query), timeout = 2 })
```

Comment:
No timeout needed, let the user time to digest its disappointment, look at the query term he entered, etc...